sky/jobs/state.py

```diff
@@ -1401,56 +1575,80 @@ def get_num_alive_jobs() -> int:
                 ]))).fetchone()[0]
 
 
-@_init_db
-def get_waiting_job() -> Optional[Dict[str, Any]]:
-    """Get the next job that should transition to LAUNCHING.
+@_init_db_async
+async def get_waiting_job(
```

Comment:
IMO all the async functions should end in `_async` for consistency, even if there is no sync version.